src/app/query/stacks/balance/stacks-ft-balances.hooks.ts

```diff
@@ -101,13 +103,20 @@ function useStacksFungibleTokenAssetBalancesUnanchoredWithMetadata(
 
 export function useStacksFungibleTokenAssetBalance(contractId: string) {
   const account = useCurrentStacksAccount();
+  const navigate = useNavigate();
   const assetBalances = useStacksFungibleTokenAssetBalancesUnanchoredWithMetadata(
     account?.address ?? ''
   );
   return useMemo(() => {
-    if (!assetBalances.length || contractId === '') return;
-    return assetBalances.find(assetBalance => assetBalance.asset.contractId.includes(contractId));
-  }, [assetBalances, contractId]);
+    const balance = assetBalances.find(assetBalance =>
+      assetBalance.asset.contractId.includes(contractId)
+    );
+    if (!balance) {
+      toast.error('Unable to find balance by contract id');
+      navigate('..');
+    }
+    return balance as StacksFungibleTokenAssetBalance;
```

Comment:
Why is this type cast needed? If we `return` in the !balance doesn't it yield the same type?

Reply:
no( it returns `StacksFungibleTokenAssetBalance | undefined`, so it causes `undefined` errors in other hooks